src/main/java/step2/domain/statistics/Rank.java

```diff
@@ -0,0 +1,64 @@
+package step2.domain.statistics;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum Rank {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    NONE(0, 0);
+
+    public int matchNumber;
+    public long winnings;
+
+    Rank(int matchNumber, long winnings) {
+        this.matchNumber = matchNumber;
+        this.winnings = winnings;
+    }
+
+    public static Rank rank(int matchCount, boolean bonusContained) {
+        if (!isContains(matchCount)) {
+            return Rank.NONE;
+        }
+
+        if (Rank.SECOND.matchNumber == matchCount && bonusContained) {
+            return Rank.SECOND;
+        }
+
+        if (Rank.THIRD.matchNumber == matchCount && !bonusContained) {
+            return Rank.THIRD;
+        }
+
+        return Arrays.stream(Rank.values())
+                .filter(r -> r.matchNumber == matchCount)
+                .findFirst()
+                .get();
```

Comment:
```suggestion
        Rank lottoRanking = Arrays.stream(Rank.values())
                .filter(rank -> rank.matchNumber == matchCount)
                .findFirst()
                .orElse(NONE);
        
        if (lottoRanking == Rank.THIRD && bonusContained) {
            lottoRanking = Rank.SECOND;
        }
        
        return lottoRanking;
```
이렇게 변경해보면 좋을 것 같아요. :)